torch/sparse/_triton_ops.py

```diff
@@ -463,19 +460,42 @@ def check(cond, msg):
                 f"but got bsr.dim() == {bsr.dim()} and dense.dim() == {dense.dim()}.",
             )
 
+            m, kl = bsr.shape[-2:]
+            kr, n = dense.shape[-2:]
+
             check(
                 kl == kr,
                 "bsr_dense_mm(): argument sizes are not compatible for matrix multiplication, "
                 f"got bsr.shape[-1] == {kl} which is not equal to dense.shape[-2] == {kr}.",
             )
 
-            row_block = bsr.values().shape[-2]
+            row_block, col_block = bsr.values().shape[-2:]
             check(
                 not n % row_block,
                 f"bsr_dense_mm(): dense.size(-1) == {n} should be divisible by "
                 f"blocksize[0] == {row_block}.",
             )
 
+            def is_power_of_two(v):
+                return not (v & (v - 1))
```

Comment:
Nit (I'm not sure how `not` is defined for integers in Python, perhaps you can send a link
```suggestion
                return (v & (v - 1)) == 0
```

Reply:
https://docs.python.org/3/reference/expressions.html#not

Reply:
My point here, is that, in my mind,  mixing logical `and` and boolean `not` hurts readability,  on the other hand, it's  up to an author/maintainer of the codebase. I.e. as long as  @cpuhrsch is fine reading that, I'm ok as well.